### PR TITLE
Add test fim with file currently open

### PR DESCRIPTION
**Pull request by CamiRomero**
|Related issue|
|---|
| #1150 |

## Description
In order to test #1150 we add a test that checks that FIM correctly work when a file is currently open

## Configuration options

Type | Format |   Tag | File name
-- | -- | -- | -- | 
Agent | msi | 4.3.0 | [wazuh-agent-4.3.0-0.40302.20211119.msi)](https://packages-dev.wazuh.com/staging/windows/wazuh-agent-4.3.0-0.40302.20211119.msi)

### test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py

| Test path |  OS  | Type  | Version  | Date | Status | Executed by | 
|---   |---    |---    |--- |--- |--- |--- |
| test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py | Agent   | Jenkins | 4.3 | 03/12/2021| [:green_circle: ](https://ci.wazuh.info/job/Test_integration/16932/) | Camila |
| test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py | Agent   | Jenkins | 4.3 |03/12/2021 | [:green_circle: ](https://ci.wazuh.info/job/Test_integration/16933/)  | Camila|
| test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py | Agent   | Jenkins | 4.3 | 03/12/2021| [:green_circle: ](https://ci.wazuh.info/job/Test_integration/16934/) | Camila|
| test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py | Agent   | Local | 4.3 |03/12/2021 |  [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7651657/r1-1150.zip) | Camila |
| test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py | Agent   | Local | 4.3 | 03/12/2021|  [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7651658/r2-1150.zip) | Camila |
| test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py | Agent   | Local | 4.3 |03/12/2021 |  [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7651659/r3-1150.zip)  | Camila |

### test_fim/

| Test path | OS  | Type  | Version | Date | Status | Executed by |
|---   |---    |---    |--- |--- |--- |--- |
| test_fim/ | Agent   | Jenkins | 4.3 | 03/12/2021 |[:green_circle: ](https://ci.wazuh.info/job/Test_integration/16942/) | Camila |
| test_fim/ | Agent   | Jenkins | 4.3 | 03/12/2021 |[:green_circle: ](https://ci.wazuh.info/job/Test_integration/16943/)  | Camila |
| test_fim/ | Agent   | Jenkins | 4.3 | 03/12/2021 |[:green_circle: ](https://ci.wazuh.info/job/Test_integration/16944/) | Camila |
| test_fim/ | Agent   | Local | 4.3 | 06/12/2021 |[:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7661176/R1-1150.zip)| Camila |
| test_fim/ | Agent   | Local | 4.3 | 06/12/2021 |[:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7661177/R2-1150.zip)  | Camila |
| test_fim/ | Agent   | Local | 4.3 | 06/12/2021 |[:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7661179/R3-1150.zip) | Camila |

## Tests

- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.

- [ ] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.

### CODE REVIEW

File: tests/integration/test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+
+# Marks
+
+pytestmark = [pytest.mark.win32, pytest.mark.tier(level=0)]
+
+
+# variables
+
+directory_str = os.path.join(PREFIX, 'testdir123')
```
Comment (damarisg) [inline tests/integration/test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py:73]:
```suggestion
directory_str = os.path.join(PREFIX, 'testdir')
```

**Review by damarisg — CHANGES_REQUESTED** [review]

File: tests/integration/test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: File Integrity Monitoring (FIM) system watches selected files and triggering alerts when these
+       files are modified. Specifically, these tests will check if FIM could work correctly when a file is open.
+       The FIM capability is managed by the 'wazuh-syscheckd' daemon, which checks configured files
+       for changes to the checksums, permissions, and ownership.
+
+tier: 0
```
[the remark this answers is not in the document]
Comment (CamiRomero) [reply tests/integration/test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py]: Done in commit [05efa847](https://github.com/wazuh/wazuh-qa/pull/2300/commits/05efa847441f749ce802d3bf9ec1fc8f024632b9)

File: tests/integration/test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+from wazuh_testing.fim import generate_params, callback_detect_integrity_event
+from wazuh_testing.tools.configuration import load_wazuh_configurations, check_apply_test
+from wazuh_testing.tools.file import write_file_without_close, remove_file
+from wazuh_testing.tools.services import control_service
+from wazuh_testing.tools import PREFIX
+
+
+# Marks
+
+pytestmark = [pytest.mark.win32, pytest.mark.tier(level=0)]
```
[the remark this answers is not in the document]
Comment (CamiRomero) [reply tests/integration/test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py]: Done in commit [05efa847](https://github.com/wazuh/wazuh-qa/pull/2300/commits/05efa847441f749ce802d3bf9ec1fc8f024632b9)

File: deps/wazuh_testing/wazuh_testing/tools/file.py [hunk 4]
```diff
@@ -120,6 +120,11 @@ def write_file(file_path, data):
         f.write(data)
 
 
+def write_file_without_close(file_path, data=''):
```
[the remark this answers is not in the document]
Comment (CamiRomero) [reply deps/wazuh_testing/wazuh_testing/tools/file.py:124]: Done in commit [cda6a3](https://github.com/wazuh/wazuh-qa/pull/2300/commits/cda6a3ec2642df071f68a64e985d595c23c3f3ce)

File: tests/integration/test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+@pytest.fixture(scope='module', params=configurations)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+# tests
+@pytest.mark.parametrize('tags_to_apply', [
+    {'ossec_conf'}
+])
```
[the remark this answers is not in the document]
Comment (CamiRomero) [reply tests/integration/test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py]: Done in commit [2ed1dde](https://github.com/wazuh/wazuh-qa/pull/2300/commits/2ed1ddef9634f062e1fbe095d3741cb150d2e892)

File: tests/integration/test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py (hunk 1 above)
Comment (CamiRomero) [reply tests/integration/test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py]: Done in commit [2ed1dde](https://github.com/wazuh/wazuh-qa/pull/2300/commits/2ed1ddef9634f062e1fbe095d3741cb150d2e892)

**Review by damarisg — APPROVED** [review]
## Test Results 

| Test Path | OS | Version | Date | Status| 
|--|--|--|--|--|
| test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py| Windows - Agent | 4.3.0 | 2021/12/06 | [🟢 ](https://ci.wazuh.info/job/Test_integration/17014/) | 
|test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py| Windows - Agent | 4.3.0 | 2021/12/06 | [🟢 ](https://ci.wazuh.info/job/Test_integration/17013/) | 
|test_fim/test_files/test_basic_usage/test_basic_usage_currently_open.py| Windows - Agent | 4.3.0 | 2021/12/06 | [🟢 ](https://ci.wazuh.info/job/Test_integration/17015/) | 
| test_fim/| Windows - Agent | 4.3.0 | 2021/12/06 | [🟢 ](https://ci.wazuh.info/job/Test_integration/17022/) | 
| test_fim/ | Windows - Agent | 4.3.0 | 2021/12/06 | [🟢 ](https://ci.wazuh.info/job/Test_integration/17021/) | 
| test_fim/| Windows - Agent | 4.3.0 | 2021/12/06 | [🟢 ](https://ci.wazuh.info/job/Test_integration/17023/) | 
| test_fim/| Macos- Agent| 4.3.0 | 2021/12/06 | [🟢 ](https://ci.wazuh.info/job/Test_integration/17018/) | 
| test_fim/| Solaris - Agent | 4.3.0 | 2021/12/06 | [🟢 ](https://ci.wazuh.info/job/Test_integration/17017/) | 
| test_fim/| Linux - Manager| 4.3.0 | 2021/12/06 | [🟢 ](https://ci.wazuh.info/job/Test_integration/17019/) | 
| test_fim/| Linux - Agent | 4.3.0 | 2021/12/06 | [🟢 ](https://ci.wazuh.info/job/Test_integration/17020/) |

**Pull request merged**